src/Mod/Sketcher/Gui/TaskSketcherGeneral.cpp

```diff
@@ -315,6 +344,12 @@ void TaskSketcherGeneral::onToggleAvoidRedundant(bool on)
     sketchView->AvoidRedundant.setValue(on);
 }
 
+void TaskSketcherGeneral::onToggleAutoUpdate(bool on)
+{
+    sketchView->getSketchObject()->noRecomputes = !on;
+    widget->saveSettings();

```

Comment:
I created saveAutoUpdate() then.
Note, Saving auto-update seems necessary or toggling it have no effect while in the sketch. Also saving it on toggle was how it was implemented in solver message.

Reply:
I can confirm that because how things are implemented, saving 'Auto update' and 'Auto remove redundant' is needed.
For 'Auto update', there is probably something to improve because the code mixes between using directly the preference parameter, and using the `noRecomputes` member var of SketchObject.